controllers/rack.go

```diff
@@ -908,8 +930,7 @@ func (r *SingleClusterReconciler) rollingRestartRack(found *appsv1.StatefulSet,
 	}
```

Comment:
at line 916, is it possible to have a pod with image failed state if Len(failed pods) is 0?